functions/APIs/deals.js

```diff
@@ -1,11 +1,14 @@
+const express = require("express");
+const dealsRoute = express.Router();
+const auth = require("../util/auth");
 const { db } = require("../util/admin");
 
 /*
 read a list of all deals
 GET /deals
 success response: array of deal objects (documented in POST /deals)
 */
-exports.getAllDeals = (request, response) => {
+dealsRoute.get("/", auth, (req, res) => {
```

Comment:
same comments as the other PRs: let's keep `request` and `response`

(also this should use single quotes, but we can defer that change, unless you want to change the entire file now for consistency. we can also defer the jsdoc updates)